binary/function.proto

```diff
@@ -105,20 +109,18 @@ message FunctionSignature {
     }
 
     message Argument {
+        ParameterizedType type = 1;
```

Comment:
Can `type` just be a `Type`? (I think I ask this same question a little further down where you define `ParameterizedType`)

Reply:
The idl for a parameterized type doesn't require any parameters to be included. So in that sense, logically yes. I guess I could make either an option here as a union but feels excessive.